src/lava/lib/optimization/problems/problems.py

```diff
@@ -177,124 +180,156 @@ def verify_solution(self, solution):
-        Eqaulity constraints offsets, by default None
-
-    Raises
-    ------
-    ValueError
-        ValueError exception raised if equality or inequality constraints
-        are not properly defined. Ex: Defining A_eq while not defining k_eq
-        and vice-versa.
-
+        Parameters
+        ----------
+        hessian : 2-D or 1-D np.array
+            Quadratic term of the cost function
+        linear_offset : 1-D np.array, optional
+            Linear term of the cost function, defaults vector of zeros of the
+            size of the number of variables in the QP
+        constraint_hyperplanes : 2-D or 1-D np.array, optional
+            Inequality constrainting hyperplanes, by default None
+        constraint_biases : 1-D np.array, optional
+            Ineqaulity constraints offsets, by default None
+        constraint_hyperplanes_eq : 2-D or 1-D np.array, optional
+            Equality constrainting hyperplanes, by default None
+        constraint_biases_eq : 1-D np.array, optional
+            Eqaulity constraints offsets, by default None
+
+        Raises
+        ------
+        ValueError
+            ValueError exception raised if equality or inequality constraints
+            are not properly defined. Ex: Defining A_eq while not defining k_eq
+            and vice-versa.
     """
 
     def __init__(
         self,
-        hessian: np.ndarray,
+        hessian: npt.ArrayLike,
         linear_offset: ty.Optional[np.ndarray] = None,
-        constraint_hyperplanes: ty.Optional[np.ndarray] = None,
-        constraint_biases: ty.Optional[np.ndarray] = None,
-        constraint_hyperplanes_eq: ty.Optional[np.ndarray] = None,
-        constraint_biases_eq: ty.Optional[np.ndarray] = None,
+        inequality_constraints_weights: ty.Optional[np.ndarray] = None,
+        inequality_constraints_biases: ty.Optional[np.ndarray] = None,
+        equality_constraints_weights: ty.Optional[np.ndarray] = None,
+        equality_constraints_biases: ty.Optional[np.ndarray] = None,
     ):
-        if (
-            constraint_hyperplanes is None and constraint_biases is not None
-        ) or (constraint_hyperplanes is not None and constraint_biases is None):
-            raise ValueError(
-                "Please properly define your Inequality constraints. Supply \
-                all A and k "
-            )
-
-        if (
-            constraint_hyperplanes_eq is None
-            and constraint_biases_eq is not None
-        ) or (
-            constraint_hyperplanes_eq is not None
-            and constraint_biases_eq is None
-        ):
-            raise ValueError(
-                "Please properly define your Equality constraints. Supply \
-                all A_eq and k_eq."
-            )
+        super().__init__()
+        self._variables._continuous = ContinuousVariables(
+            num_variables=hessian.shape[0]
+        )
+        self.q_cost = Cost(linear_offset, hessian)

```

Comment:
Not sure if you want to add a getter/setter for q_cost, and use self._q_cost instead.
Up to you.

Reply:
private variables shouldn't be accessed like that. Not good practice